src/servers/sapphire_zone/Zone/TerritoryMgr.cpp

```diff
@@ -161,8 +161,9 @@ Core::ZonePtr Core::TerritoryMgr::createTerritoryInstance( uint32_t territoryTyp
    auto pZone = make_Zone( territoryTypeId, getNextInstanceId(), pTeri->name, pPlaceName->name );
    pZone->init();
 
-   m_territoryInstanceMap[pZone->getTerritoryId()][pZone->getGuId()] = pZone;
    m_instanceIdToZonePtrMap[pZone->getGuId()] = pZone;
+   m_territoryInstanceMap[pZone->getTerritoryId()][pZone->getGuId()] = pZone;
+   m_zoneSet.insert( { pZone } );
```

Comment:
If i am not mistaken, this would result in zonelogic being processed twice.
instance and zone are handled seperatly.

Reply:
nah, `m_instanceZoneSet` is responsible for the instance update loop whereas `m_zoneSet` is for normal zones for the same purpose. `m_territoryInstanceMap` is just so you can get all instances of a specific territory, not sure if we actually use that for anything at the moment though

Reply:
Right, brainfart and ambigious naming.

Reply:
yeah, naming isn't very good here, perhaps rename it to `m_zoneUpdateSet` and `m_instanceUpdateSet`?

Reply:
not sure if that's any better though lmao